src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -1831,19 +1909,40 @@ internal static class Nullable_Delegates
         public unsafe delegate int GetValueDelegate(IntPtr thisPtr, IntPtr* value);
         public unsafe delegate int GetValueDelegateAbi(void* thisPtr, void* value);
     }
+
+    internal static class IReferenceIIDs
+    {
+#if NET
+        internal static readonly Guid IReferenceMatrix3x2_IID = new(new ReadOnlySpan<byte>(new byte[] { 0xfd, 0x8c, 0x35, 0x76, 0xbd, 0x2c, 0x5b, 0x52, 0xa4, 0x9e, 0x90, 0xee, 0x18, 0x24, 0x7b, 0x71 }));
+        internal static readonly Guid IReferenceMatrix4x4_IID = new(new ReadOnlySpan<byte>(new byte[] { 0xdc, 0xff, 0xcb, 0xda, 0xef, 0x68, 0xd0, 0x5f, 0xb6, 0x57, 0x78, 0x2d, 0x0a, 0xc9, 0x80, 0x7e }));
+        internal static readonly Guid IReferencePlane_IID = new(new ReadOnlySpan<byte>(new byte[] { 0xa1, 0x42, 0xd5, 0x46, 0xf7, 0x52, 0xe7, 0x58, 0xac, 0xfc, 0x9a, 0x6d, 0x36, 0x4d, 0xa0, 0x22 }));
+        internal static readonly Guid IReferenceQuaternion_IID = new(new ReadOnlySpan<byte>(new byte[] { 0xbb, 0x04, 0x70, 0xb2, 0x14, 0xc0, 0xce, 0x5d, 0x9a, 0x21, 0x79, 0x9c, 0x5a, 0x3c, 0x14, 0x61 }));
+        internal static readonly Guid IReferenceVector2_IID = new(new ReadOnlySpan<byte>(new byte[] { 0x9e, 0xa6, 0xf6, 0x48, 0x65, 0x84, 0xae, 0x57, 0x94, 0x00, 0x97, 0x64, 0x08, 0x7f, 0x65, 0xad }));
+        internal static readonly Guid IReferenceVector3_IID = new(new ReadOnlySpan<byte>(new byte[] { 0xff, 0x70, 0xe7, 0x1e, 0x54, 0xc9, 0xca, 0x59, 0xa7, 0x54, 0x61, 0x99, 0xa9, 0xbe, 0x28, 0x2c }));
+        internal static readonly Guid IReferenceVector4_IID = new(new ReadOnlySpan<byte>(new byte[] { 0xc9, 0x43, 0xe8, 0xa5, 0x20, 0xed, 0x39, 0x53, 0x8f, 0x8d, 0x9f, 0xe4, 0x04, 0xcf, 0x36, 0x54 }));
+#else
+        internal static readonly Guid IReferenceMatrix3x2_IID = new(0x76358cfd, 0x2cbd, 0x525b, 0xa4, 0x9e, 0x90, 0xee, 0x18, 0x24, 0x7b, 0x71);
+        internal static readonly Guid IReferenceMatrix4x4_IID = new(0xdacbffdc, 0x68ef, 0x5fd0, 0xb6, 0x57, 0x78, 0x2d, 0x0a, 0xc9, 0x80, 0x7e);
+        internal static readonly Guid IReferencePlane_IID = new(0x46d542a1, 0x52f7, 0x58e7, 0xac, 0xfc, 0x9a, 0x6d, 0x36, 0x4d, 0xa0, 0x22);
+        internal static readonly Guid IReferenceQuaternion_IID = new(0xb27004bb, 0xc014, 0x5dce, 0x9a, 0x21, 0x79, 0x9c, 0x5a, 0x3c, 0x14, 0x61);
+        internal static readonly Guid IReferenceVector2_IID = new(0x48f6a69e, 0x8465, 0x57ae, 0x94, 0x00, 0x97, 0x64, 0x08, 0x7f, 0x65, 0xad);
+        internal static readonly Guid IReferenceVector3_IID = new(0x1ee770ff, 0xc954, 0x59ca, 0xa7, 0x54, 0x61, 0x99, 0xa9, 0xbe, 0x28, 0x2c);
+        internal static readonly Guid IReferenceVector4_IID = new(0xa5e843c9, 0xed20, 0x5339, 0x8f, 0x8d, 0x9f, 0xe4, 0x04, 0xcf, 0x36, 0x54);
+#endif
+    }
 }
 
 #if NET
-namespace WinRT
+    namespace WinRT
 {
     public sealed class StructTypeDetails<T, TAbi> : IWinRTExposedTypeDetails where T: struct where TAbi : unmanaged
     {
         private static readonly Guid PIID = GuidGenerator.CreateIID(typeof(ABI.System.Nullable<T>));
```

Comment:
We could hardcode all the well known IIDs here as well to avoid the call to `GuidGenerator`. We can skip this in this PR though since I'm doing the same in #1429, so might as well just fix both occurrences there.

Reply:
Lets defer on this discussion to the other PR where I will be responding to the nulllable changes soon.